### ISSUE TITLE
Export to glTF 2.0 as a single object for performance

### ISSUE THREAD

**Issue by andreasrosdal** [KhronosGroup/glTF-Blender-Exporter#76]
This is a feature-request to allow exporting a complex object consisting of multiple object to a single joined object in glTF 2.0. I'm using this exporter to export glTF 2.0 models for [Freeciv WebGL](https://github.com/freeciv/freeciv-web), and it seems that the performance is better when the models are exported as a single object. This could be similar to the Object -> Join menu item, before the export to glTF 2.0.

**Comment by UX3D-nopper** [comment]
Agree, and for the future, we hope we can integrate performance critical functions directly into Blender, which can be executed in C/C++.

Think this could be easily implemented. Will have a look at it on Monday.

**Comment by UX3D-nopper** [comment]
I did have a look at it and these two lines do the job:
```
bpy.ops.object.select_all(action = 'SELECT')
bpy.ops.object.join()

```

The problem is, that the whole scene has been modified afterwards. To solve this and - preserve the original state of the scene - copies need to be created but with its original name. And this chain ends up until the material.

I will postpone this for now, as other issues seem to be more important, as this can be solved easily by a manual process.

**Comment by andreasrosdal** [comment]
Thanks for looking at this, I hope it will be given priority at some point.

**Comment by UX3D-nopper** [comment]
Thank u for honoring this.
As the Blender glTF 2.0 exporter is open source, we hope that more people will participate on implementing features.

**Comment by capitalknew** [comment]
Although my rendering is not very fast, 200,000 polygons can hardly endure (the same sponza), not to mention the millions where only 3 fps from strength. This is real-time path tracing.

**Comment by UX3D-nopper** [comment]
I do not understand.
In Cycles, yes, the whole scene is using real-time path tracing. But as soon as you export the scene and render it, you ar using the real-timer PBR approach, as seen in UE4 and/or Unity.

**Comment by donmccurdy** [comment]
Would this make more sense as part of https://github.com/AnalyticalGraphicsInc/gltf-pipeline? Or would it be harder to implement after export?
